eth/downloader/receiptreference_test.go

```diff
@@ -0,0 +1,105 @@
+package downloader
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/assert"
+)
+
+// makeCorrection is a helper function to create a slice of receipts and a slice of corrected receipts
+func makeCorrection(bn uint64, cid uint64, ns []uint64, ty []uint8) (types.Receipts, types.Receipts) {
+	receipts := make(types.Receipts, len(ns))
+	correctedReceipts := make(types.Receipts, len(ns))
+	transactions := make(types.Transactions, len(ns))
+	for i := range ns {
+		receipts[i] = &types.Receipt{Type: ty[i], DepositNonce: &ns[i]}
+		correctedReceipts[i] = &types.Receipt{Type: ty[i], DepositNonce: &ns[i]}
+		transactions[i] = types.NewTx(&types.DepositTx{})
+	}
+
+	correctedReceipts = correctReceipts(correctedReceipts, transactions, bn, cid)
+
+	return receipts, correctedReceipts
+}
+
+func TestCorrectReceipts(t *testing.T) {
+	type testcase struct {
+		bn       uint64
```

Comment:
These short names are kinda hard to understand. Even a comment would go a long way if you don't want to make the names longer.